test/aggregate.js

```
@@ -1,9 +1,173 @@
+  t.plan(5)
+  const options = _.cloneDeep(classBreaks)
+  options.classificationDef.classificationField = 'House_Number'
+  options.classificationDef.normalizationType = 'esriNormalizeByField'
+  options.classificationDef.normalizationField = 'Trunk_Diameter'
+  const results = winnow.query(features, options)
+  t.equal(Array.isArray(results), true)
+  t.equal(Array.isArray(results[0]), true)
+  t.equal(results.length, 5)
+  t.deepEqual(results[0], [0.07142857142857142, 910.0571428571428])
+  t.deepEqual(results[4], [3640.0142857142855, 4550])
+})
+
+test('normalize by log', t => {
+  t.plan(5)
+  const options = _.cloneDeep(classBreaks)
+  options.classificationDef.normalizationType = 'esriNormalizeByLog'
+  const results = winnow.query(features, options)
+  t.equal(Array.isArray(results), true)
+  t.equal(Array.isArray(results[0]), true)
+  t.equal(results.length, 5)
+  t.deepEqual(results[0], [0, 0.2292256071356476])
+  t.deepEqual(results[4], [0.9169024285425906, 1.146128035678238])
+})
+
+test('normalize by total', t => {
+  t.plan(5)
+  const options = _.cloneDeep(classBreaks)
+  options.classificationDef.normalizationType = 'esriNormalizeByPercentOfTotal'
+  const results = winnow.query(features, options)
+  t.equal(Array.isArray(results), true)
+  t.equal(Array.isArray(results[0]), true)
+  t.equal(results.length, 5)
+  t.deepEqual(results[0], [0, 0.0013272469579973742])
+  t.deepEqual(results[4], [0.005308987831989498, 0.006636234789986871])
+})
+
+test('unacceptable classificationField', t => {
+  t.plan(1)
+  const options = _.cloneDeep(classBreaks)
+  options.classificationDef.classificationField = 'Common_Name'
+  const results = winnow.query(features, options)
+  t.equal(results, undefined)
+})
+
+test('unacceptable classificationMethod', t => {
+  t.plan(1)
+  const options = _.cloneDeep(classBreaks)
+  options.classificationDef.classificationMethod = 'invalidMethod'
+  const results = winnow.query(features, options)
```

this should throw an error